maven/impl-maven-embedded/src/main/java/org/jboss/shrinkwrap/resolver/impl/maven/embedded/pom/equipped/ConfigurationStageImpl.java

```diff
@@ -80,12 +80,13 @@ public ConfigurationDistributionStage setLogger(InvokerLogger invokerLogger) {
 
     @Override
     public ConfigurationDistributionStage setWorkingDirectory(File workingDirectory) {
+        getInvocationRequest().setBaseDirectory(workingDirectory);
```

Comment:
If I understand this correctly - this is a fix for the case when a user wants to set the root directory where the `pom.xml` file is placed. The `setAlternatePomFile` method provides the option of specifying the whole path to some specific pom file, right?

Reply:
You could show the differences in a test

Reply:
The fix is for the workingDirectory not getting properly set, as rightly pointed by you because 
of  [this](http://svn.apache.org/viewvc/maven/shared/tags/maven-invoker-3.0.0/src/main/java/org/apache/maven/shared/invoker/MavenCommandLineBuilder.java?revision=1779850&view=markup#l351). Which results in always setting the workingDirectory from the parent Pom. 

The difference between `setWorkingDirectory` and `setAlternatePom`, in case of PomEquippedEmbeddedMavenImpl is

- setting workingDirectory only has the effect of changing the directory and the pom file remains the same as the one set previously.
- setting alternate pom file (setting -f), in addition to changing the working directory also, has the effect of resetting the pom file with a new one.